mafContrib/transientAsciiSEDMetric.py

```diff
@@ -0,0 +1,662 @@
+            for obs_time in flt_times:
+                filter_mag.append(
+                    redshifted_model.bandmag("lsst" + flt, "ab", obs_time)
+                )
+            # Set light_curve_mags for array indices corresponding to observations of the
+            # current filter.
+            light_curve_mags[np.where(filters == flt)[0]] = np.array(filter_mag)
+            self.light_curve_mags = light_curve_mags
+
+    def evaluate_all_detection_criteria(self, dataSlice):
+        """
+        Wrapper function to setup loop for each transient light curve and
+        evaluate all detection criteria.
+
+        Parameters:
+        -----------
+            dataSlice : numpy.array
+                Numpy structured array containing the data related to the visits
+                provided by the slicer.
+
+        Returns (to self):
+        ------------------
+            transient_detected: np.array, boolean
+                Array containing boolean tracking variable whether transient is
+                detected by passing all criteria.
+            num_detected: int
+                Scalar value of the number of transients that were detected in
+                total between all phase shifts considerd.
+
+        """
+        # Track whether each individual light curve was detected.
+        # Start with the assumption that it is True, and if it fails
+        # criteria then becomes False.
+        self.transient_detected = np.ones(len(np.unique(self.transient_id)), dtype=bool)
+
+        # Loop through each lightcurve and check if it meets requirements.
+        for i, (start_ind, end_ind) in enumerate(
+            zip(self.transient_start_index, self.transient_end_index)
+        ):
+            t_id = i
+            while self.transient_detected[t_id] == True:
+                # If there were no observations at all for this lightcurve:
+                if start_ind == end_ind:
+                    self.transient_detected[t_id] = False
+
+                self.observation_epoch_above_thresh = self.observation_epoch[
+                    start_ind:end_ind
+                ][np.where(self.obs_above_SNR_threshold[start_ind:end_ind])]
+
+                self.evaluate_pre_time_detection_criteria(t_id)
+                self.evaluate_phase_section_detection_criteria(t_id)
+                self.evaluate_number_filters_detection_criteria(
+                    dataSlice, start_ind, end_ind, t_id
+                )
+                self.evaluate_filter_in_time_detection_criteria(t_id)
+                # Finished with current set of conditions
+
+                # If detected is still true break while loop and continue to
+                # next transient
+                break
+
+        # Find the unique number of light curves that passed the required
+        # number of conditions
+        self.num_detected += len(np.where(self.transient_detected == True)[0])
+
+    def evaluate_pre_time_detection_criteria(self, t_id):
```

Comment:
I like how these criteria are broken out into separate methods - it makes it very easy to see the individual criteria and will make it very easy to modify one of them without worrying about effect on the others (or turn a single criteria off, etc.).